diff --git a/src/Plugin/QuickContent/QuickBlockContent.php b/src/Plugin/QuickContent/QuickBlockContent.php
index 5ab47b6..457baa5 100644
--- a/src/Plugin/QuickContent/QuickBlockContent.php
+++ b/src/Plugin/QuickContent/QuickBlockContent.php
@@ -138,7 +138,7 @@ class QuickBlockContent extends QuickContent implements QuicktabContentInterface
     $rids = array_keys($user->roles);
 
     // Get authorized rids.
-    $authorized_rids = db_select('block_role', 'br')
+    $authorized_rids = \Drupal::database()->select('block_role', 'br')
       ->fields('br', array('rid'))
       ->condition('module', $module, '=')
       ->condition('delta',  $delta,  '=')
