diff --git a/src/QuicktabListBuilder.php b/src/QuicktabListBuilder.php
index c9b2536..d8b2d91 100644
--- a/src/QuicktabListBuilder.php
+++ b/src/QuicktabListBuilder.php
@@ -51,22 +51,22 @@ class QuicktabListBuilder extends ConfigEntityListBuilder {
       $operations['edit'] = array(
         'title' => t('Edit quicktab'),
         'weight' => 10,
-        'url' => $entity->urlInfo('edit'),
+        'url' => $entity->toUrl('edit'),
       );
       $operations['delete'] = array(
         'title' => t('Delete quicktab'),
         'weight' => 20,
-        'url' => $entity->urlInfo('delete'),
+        'url' => $entity->toUrl('delete'),
       );
       $operations['clone'] = array(
         'title' => t('Clone quicktab'),
         'weight' => 30,
-        'url' => $entity->urlInfo('clone'),
+        'url' => $entity->toUrl('clone'),
       );
       $operations['export'] = array(
         'title' => t('Export quicktab'),
         'weight' => 40,
-        'url' => $entity->urlInfo('export'),
+        'url' => $entity->toUrl('export'),
       );
     }
     return $operations;
